xmtp_mls/src/groups/mod.rs

```diff
@@ -223,6 +245,36 @@ where
 
                 Ok((commit_bytes, post_commit_data))
             }
+            IntentKind::RemoveMembers => {
+                let intent_data = RemoveMembersIntentData::from_bytes(intent.data.as_slice())?;
+                let leaf_nodes: Vec<LeafNodeIndex> = openmls_group
+                    .members()
+                    .filter(|member| intent_data.installation_ids.contains(&member.signature_key))
+                    .map(|member| member.index)
+                    .collect();
+
+                let num_leaf_nodes = leaf_nodes.len();
+
+                if num_leaf_nodes != intent_data.installation_ids.len() {
+                    return Err(GroupError::Generic(format!(
+                        "expected {} leaf nodes, found {}",
+                        intent_data.installation_ids.len(),
+                        num_leaf_nodes
+                    )));
+                }
+
+                // The second return value is a Welcome, which is only possible if there
```

Comment:
It might make sense to have a utility method that just removes all pending proposals, using [this](https://openmls.tech/openmls/doc/openmls/group/struct.MlsGroup.html#method.remove_pending_proposal)

Reply:
100% on board